projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js

```diff
@@ -6,220 +6,285 @@ import { Placeholder, Button, Spinner } from '@wordpress/components';
+	// This let's the component know if the content was set (basically on first pass).
+
+	useEffect(
+		() => {
+			// That will only happen once
+			if ( showAnimation ) {
+				// This is to animate text input. I think this will give an idea of a "better" AI.
+				// At this point this is an established pattern.
+				const tokens = html.split( ' ' );
+				for ( let i = 1; i < tokens.length; i++ ) {
+					const output = tokens.slice( 0, i ).join( ' ' );
+					setTimeout( () => setDisplayedRawHTML( output ), 50 * i );
+				}
+				setTimeout( () => {
+					setDisplayedRawHTML( html );
+					onAnimationDone();
+				}, 50 * tokens.length );
+			} else {
+				setDisplayedRawHTML( html );
 			}
-			setTimeout( () => setAttributes( { content: content } ), 50 * tokens.length );
-		} )
-		.catch( () => {
-			setErrorMessage(
-				__(
-					'Whoops, we have encountered an error. AI is like really, really hard and this is an experimental feature. Please try again later.',
-					'jetpack'
-				)
-			);
-			setLoadingCompletion( false );
-		} );
+		},
+		// eslint-disable-next-line
+		[]
+	);
+
+	return (
+		<div className="content">
+			<RawHTML>{ displayedRawHTML }</RawHTML>
+		</div>
+	);
 }
 
-function createPrompt( title = '', content = '', categoryNames = '' ) {
-	// If title is not added, we will only complete.
-	if ( ! title ) {
-		return content;
-	}
+export default function Edit( { attributes, setAttributes, clientId } ) {
+	const [ isLoadingCompletion, setIsLoadingCompletion ] = useState( false );
+	const [ isLoadingCategories, setIsLoadingCategories ] = useState( false );
+	const [ isWaitingForPreviousBlock, setIsWaitingForPreviousBlock ] = useState( false );
+	const [ needsMoreCharacters, setNeedsMoreCharacters ] = useState( 0 );
+	const [ showRetry, setShowRetry ] = useState( false );
+	const [ errorMessage, setErrorMessage ] = useState( false );
 
-	// Title, content and categories
-	if ( content && categoryNames.length ) {
-		return sprintf(
-			/** translators: This will be a prompt to OpenAI to generate a post based on the post title, comma-seperated category names and the last 240 characters of content. */
-			__( "This is a post titled '%1$s', published in categories '%2$s':\n\n … %3$s", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
-			title,
-			categoryNames,
-			content
-		);
-	}
+	// Let's grab post data so that we can do something smart.
+	const currentPostTitle = useSelect( select =>
+		select( 'core/editor' ).getEditedPostAttribute( 'title' )
+	);
 
-	// No content, only title and categories
-	if ( categoryNames.length ) {
-		return sprintf(
-			/** translators: This will be a prompt to OpenAI to generate a post based on the post title, and comma-seperated category names. */
-			__( "This is a post titled '%1$s', published in categories '%2$s':\n", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
-			title,
-			categoryNames
-		);
-	}
+	let loading = false;
```

Comment:
What do you think about moving this closer to where it is used ( maybe on line 66 ) ?